components/clarinet-format/src/formatter/mod.rs

```diff
@@ -1405,6 +1412,19 @@ mod tests_formatter {
         assert_eq!(result, expected);
     }
 
+    #[test]
+    fn top_level_exprs() {
+        let src = r#"(let ((x (+ u1 u1)))
+  (map-insert ns x true)
+)
+(define-public (get-value)
+  (ok (map-get? ns u2))
+)
```

Comment:
Shouldn't it be
```clarity
(let ((x (+ u1 u1)))
  (map-insert ns x true)
)

(define-public (get-value)
  (ok (map-get? ns u2))
)

```

With an extra new-line before define-public?

Reply:
That is part of this issue: https://github.com/hirosystems/clarinet/issues/1769

The problem there was the top-comments get in the way of easily new-lining define- functions